test/passing/dune

```diff
@@ -16,14 +17,34 @@ let read_lines fn =
 let register_file tests fname =
   match String.split_on_char '.' fname with
   | test_name :: (("ml" | "mli") as ext) :: rest -> (
-      let test_name = test_name ^ "." ^ ext in
+      let src_test_name = test_name ^ "." ^ ext in
       let setup =
-        match Hashtbl.find tests test_name with
+        match Hashtbl.find tests src_test_name with
         | setup -> setup
-        | exception Not_found ->
-            let s = {has_ref= false; has_opts= false; extra_deps= []} in
-            Hashtbl.add tests test_name s ;
-            s
+        | exception Not_found -> (
+          (* foo_file-some_variant.ml should derive from foo_file.ml *)
+          match String.index_opt test_name '-' with
+          | None ->
+              let s =
+                { has_ref= false
+                ; has_opts= false
+                ; base_file= None
+                ; extra_deps= [] }
+              in
+              Hashtbl.add tests src_test_name s ;
+              s
```

Comment:
You could move that to a `add_test ?base_file src_test_name` function to avoid duplication.

Reply:
Done, I also re-enabled the `.mlt` tests (don't know since when they are disabled)